src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -1068,5 +1068,25 @@ public sealed override bool Equals(Symbol other, TypeCompareKind compareKind)
                 nps.Equals(this, compareKind) :
                 base.Equals(other, compareKind);
         }
+
+#nullable enable
+        internal DiagnosticInfo? DeriveCompilerFeatureRequiredDiagnostic(MetadataDecoder decoder)
+            => PEUtilities.DeriveCompilerFeatureRequiredAttributeDiagnostic(this, (PEModuleSymbol)ContainingModule, Handle, allowedFeatures: CompilerFeatureRequiredFeatures.None, decoder);
+
+        public override bool HasUnsupportedMetadata
+        {
+            get
+            {
+                var containingModule = (PEModuleSymbol)ContainingModule;
+                var decoder = ContainingSymbol switch
+                {
+                    PEMethodSymbol method => new MetadataDecoder(containingModule, method),
+                    PEPropertySymbol property => new MetadataDecoder(containingModule, (PENamedTypeSymbol)ContainingType),
+                    _ => throw ExceptionUtilities.UnexpectedValue(this.ContainingSymbol.Kind)
+                };
+
+                return DeriveCompilerFeatureRequiredDiagnostic(decoder) is { Code: (int)ErrorCode.ERR_UnsupportedCompilerFeature } || base.HasUnsupportedMetadata;
```

Comment:
>ERR_UnsupportedCompilerFeature [](http://example.com/codeflow?start=98&length=30)

Do we need to check the specific error code, or does null vs. non-null suffice?

Reply:
It would probably suffice to use non-null as the check, but I prefer to be specific as to the intent of the check. The intent is specifically to look for the unsupported compiler feature diagnostic, so that's what I'm doing. I'll adjust a couple of places where this isn't the intent.